src/components/bar/modules/updates/index.tsx

```diff
@@ -21,8 +22,10 @@ const {
 
 const pendingUpdates: Variable<string> = Variable('0');
 const postInputUpdater = Variable(true);
+const isVis = Variable(!autoHide.get());
 
 const processUpdateCount = (updateCount: string): string => {
+    isVis.set(!autoHide.get() || (autoHide.get() && parseFloat(updateCount) > 0));
```

Comment:
Let's move this down to the `updatesIcon` Variables.derive scope since this doesn't trigger on update.

Reply:
do you mean here?

```
const updatesIcon = Variable.derive(
    [bind(icon.pending), bind(icon.updated), bind(pendingUpdates)],
    (pendingIcon, updatedIcon, pUpdates) => {
        // new add below;
        isVis.set(!autoHide.get() || (autoHide.get() && parseFloat(pUpdates) > 0));
        return parseFloat(pUpdates) === 0 ? updatedIcon : pendingIcon;
    },
);
```

Reply:
Yep

Reply:
please check below commit, I move the code from your suggestion